app/classes/knight.py

```diff
@@ -1,22 +1,17 @@
 class Knight:
     def __init__(self, knight: dict) -> None:
-        self.name = knight["name"]
-        self.power = knight["power"]
-        self.hp = knight["hp"]
-        self.armour = knight["armour"]
-        self.weapon = knight["weapon"]
-        self.potion = knight["potion"]
+        self.name = knight.get("name")
+        self.power = knight.get("power")
+        self.hp = knight.get("hp")
+        self.armour = knight.get("armour")
+        self.weapon = knight.get("weapon")
+        self.potion = knight.get("potion")
```

Comment:
I'd prefer something more explicit, like knight_dict, since it's not an object atm, so it's like same object in one scope which can sometime trick you.